src/Input/Input.js

```diff
@@ -182,5 +247,28 @@ module.exports = kind(
 	*/
 	down: function () {
 		return false;
+	},
+
+	/**
+	* @private
+	*/
+	minChanged: function () {
+		this.setAttribute('min', this.min);
+	},
+
+	/**
+	* @private
+	*/
+	maxChanged: function () {
+		this.setAttribute('max', this.max);
+	},
+
+	/**
+	* @private
+	*/
+	invalidChanged: function () {
+		this.addRemoveClass('input-invalid', this.invalid);
+		this.$.tooltip.activator = this;
+		this.$.tooltip.set('showing', this.invalid && this.invalidMessage && this.invalidMessage !== '');
```

Comment:
I don't think the last part of this statement is necessary, as `this.invalidMessage` will be falsy if the value is `''`.